### PR TITLE
chore: merge repo & doc-engine into one before cleaning up

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Merges the doc service repo and doc engine in order the cleanup the query processing code 

### How to test it?

Check it works exactly the same as before

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/core/core/src/services/document-service/repository.ts [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+        assoc('publishedAt', new Date()),
+        assoc('documentId', documentId),
+        omit('id'),
+        // Transform relations to target published versions
+        (entry) => {
+          const opts = { uid, locale: entry.locale, isDraft: false, allowMissingId: true };
+          return transformData(entry, opts);
+        },
+        // Create the published entry
+        (data) => create({ ...queryParams, data, locale: data.locale })
```
Comment (Marc-Roig) [inline packages/core/core/src/services/document-service/repository.ts:326]: won't this conflict with the create always creating a draft by defualt?

Comment (alexandrebodin) [reply packages/core/core/src/services/document-service/repository.ts:326]: definitely, I should have created in Draft It's not ready yet :)

File: packages/core/utils/src/convert-query-params.ts [hunk 2]
```diff
@@ -610,7 +610,7 @@ const transformParamsToQuery = (uid: string, params: Params): Query => {
 
   const query: Query = {};
 
-  const { _q, sort, filters, fields, populate, page, pageSize, start, limit } = params;
+  const { _q, sort, filters, fields, populate, page, pageSize, start, limit, ...rest } = params;
```
Comment (Marc-Roig) [inline packages/core/utils/src/convert-query-params.ts:613]: which other params can be sent ?

**Review by Marc-Roig — APPROVED** [review]

File: packages/core/utils/src/convert-query-params.ts (hunk 2 above)
Comment (alexandrebodin) [reply packages/core/utils/src/convert-query-params.ts:613]: it's to allow chaining with the other functions (data, lookup at the moment)

**Pull request merged**